File: 0x08-python-more_classes/3-rectangle.py
#!/usr/bin/python3
"""Defines a Rectangle class."""


class Rectangle:
    """Represent a rectangle."""

    def __init__(self, width=0, height=0):
        self.width = width
        self.height = height

    @property
    def width(self):
        return self.__width

    @width.setter
    def width(self, value):
        '''Setting up the value and width'''
        if not isinstance(value, int):
            raise TypeError("width must be an integer")
        if value < 0:
            raise ValueError("width must be >= 0")
        self.__width = value

    @property
    def height(self):
        return self.__height

    @height.setter
    def height(self, value):
        '''Setting up the value and height'''
        if not isinstance(value, int):
            raise TypeError("height must be an integer")
        if value < 0:
            raise ValueError("height must be >= 0")
        self.__height = value

    def area(self):
        '''for displaying the area'''
        return self.__width * self.__height

    def perimeter(self):
        '''a function for returning the perimeter'''
        if self.__width == 0 or self.__height == 0:
            return 0
        return (self.__height + self.__width) * 2

    def __str__(self):
        '''function to create a rectangle'''
        return '\n'.join(['#' * self.__width] * self.__height)

    def printing(self):
        '''for displaying the created rectangle'''
        print(str(self))

File: 0x13-javascript_objects_scopes_closures/3-rectangle.js
#!/usr/bin/node
class Rectangle {
  constructor (w, h) {
    if (w > 0 && h > 0) {
      this.width = w;
      this.height = h;
    }
  }

  print () {
    for (let i = 0; i < this.height; i++) {
      let sn = '';
      for (let g = 0; g < this.width; g++) {
        sn += 'x';
      }
      console.log(sn);
    }
  }
}
module.exports = Rectangle;
